sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/policy/ResponseValidationPolicyBuilder.java

```diff
@@ -47,12 +54,22 @@ public HttpPipelinePolicy build() {
      * @deprecated Use {@link  #addOptionalEcho(HttpHeaderName)} instead.
      */
     @Deprecated
+    @SuppressWarnings("unchecked")
     public ResponseValidationPolicyBuilder addOptionalEcho(String headerName) {
-        assertions.add((httpResponse, logger) -> {
+        assertions.add((httpResponse, logger, context) -> {
             HttpHeaderName httpHeaderName = HttpHeaderName.fromString(headerName);
             String requestHeaderValue = httpResponse.getRequest().getHeaders().getValue(httpHeaderName);
             String responseHeaderValue = httpResponse.getHeaders().getValue(httpHeaderName);
-            if (responseHeaderValue != null && !responseHeaderValue.equals(requestHeaderValue)) {
+
+            List<HttpHeaderName> headersToSkip = new ArrayList<>();
+            Optional<Object> contextAdjustment = context.getData(Constants.SKIP_ECHO_VALIDATION_KEY);
+            if (contextAdjustment.isPresent()) {
+                headersToSkip = (List<HttpHeaderName>) contextAdjustment.get();
+            }
```

Comment:
It would probably makes sense to break this into its own private helper method (e.g. `getHeadersToSkip`) so that we can reuse it across both versions of this `addOptionalEcho` and so we can narrow the suppression to just this logic instead of the entire `addOptionalEcho` methods.